[tempo-distributor] Add externalTrafficPolicy option to service

celian-garcia opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/helm-charts?pullRequest=3182) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/helm-charts?pullRequest=3182) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/helm-charts?pullRequest=3182) it.</sub>

celian-garcia commented:
Hello, I updated the PR to fix the lint issues and resolve conflicts. Please can you run again the checks and review? 

Here is the command I ran
```bash
$ podman run --rm --volume "$(pwd):/helm-docs" -u "$(id -u)" jnorwood/helm-docs:v1.8.1
time="2024-07-08T12:13:41Z" level=info msg="Found Chart directories [.]"
time="2024-07-08T12:13:41Z" level=info msg="Generating README Documentation for chart /helm-docs"
```

celian-garcia commented:
ping @mapno

Sheikh-Abubaker commented on charts/tempo-distributed/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 description: Grafana Tempo in MicroService mode
 type: application
-version: 1.14.0
+version: 1.14.1
```
```suggestion
version: 1.15.0
```

Sheikh-Abubaker commented on charts/tempo-distributed/values.yaml:
```diff
@@ -465,6 +465,8 @@ distributor:
     # -- If type is LoadBalancer limit incoming traffic from IPs.
     loadBalancerSourceRanges: []
+    # -- If type is LoadBalancer you can set it to 'Local' [preserve the client source IP](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)
+    externalTrafficPolicy: null
```
```suggestion
    externalTrafficPolicy: {}
```

Sheikh-Abubaker commented on charts/tempo-distributed/values.yaml:
```diff
@@ -465,6 +465,8 @@ distributor:
     # -- If type is LoadBalancer limit incoming traffic from IPs.
     loadBalancerSourceRanges: []
+    # -- If type is LoadBalancer you can set it to 'Local' [preserve the client source IP](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)
+    externalTrafficPolicy: null
```
```suggestion
    externalTrafficPolicy: {}
```

celian-garcia replied:
Thanks for the review! I'm not sure about this one the field is a string and not an object so I applied what seemed to be a policy to me. 
![image](https://github.com/grafana/helm-charts/assets/10258608/b38b659f-2847-4574-8fab-77b9ad8dd514)

Sheikh-Abubaker replied:
You want  `externalTrafficPolicy` to be configurable right ?

celian-garcia replied:
Yes exactly, or if they want to put nothing, it just relies on the default kubernetes value. Field not set

Sheikh-Abubaker replied:
> if they want to put nothing,

That's what we are doing when we set  `externalTrafficPolicy: {}` it is an empty object and user configurable.

celian-garcia replied:
Having the default value to ``nil`` or ``{}`` will have the same effect. To not pass inside my ``with``. 
The thing is I noted that all the other default of strings are ususally ``nil`` and default of objects are usually ``{}``. I'm just following it for consistency.

Sheikh-Abubaker replied:
> Having the default value to `nil` or `{}` will have the same effect.

That's right!!

Sheikh-Abubaker commented on charts/tempo-distributed/README.md:
```diff
@@ -1,6 +1,6 @@
 # tempo-distributed
 
-![Version: 1.14.0](https://img.shields.io/badge/Version-1.14.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.5.0](https://img.shields.io/badge/AppVersion-2.5.0-informational?style=flat-square)
+![Version: 1.14.1](https://img.shields.io/badge/Version-1.14.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.5.0](https://img.shields.io/badge/AppVersion-2.5.0-informational?style=flat-square)
```
```suggestion
![Version: 1.15.0](https://img.shields.io/badge/Version-1.15.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.5.0](https://img.shields.io/badge/AppVersion-2.5.0-informational?style=flat-square)
```

Sheikh-Abubaker commented:
Please address the changes, apart from this, LGTM!

celian-garcia commented:
> Please address the changes, apart from this, LGTM!

Thanks @Sheikh-Abubaker I updated my commit. I noticed that you added a merge commit so I just rewrote my commit rebasing myself onto the last version of main. Hope you don't mind.

Sheikh-Abubaker reviewed (approved):
LGTM!!

mapno reviewed (approved):
LGTM
